src/react/components/common/tagInput/tagInputItem.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import { ITag } from "../../../../models/applicationState";
+import { idealTextColor, elipsify } from "../../../../common/utils";
+import { GithubPicker } from "react-color";
+import { TagEditMode } from "./tagInput";
+import { constants } from "../../../../common/constants";
+// tslint:disable-next-line:no-var-requires
+const tagColors = require("../tagColors.json");
+
+export interface IVerticalTagItemProps {
+    tag: ITag;
+    index: number;
+    isLocked: boolean;
+    isSelected: boolean;
+    isBeingEdited: boolean;
+    tagEditMode: TagEditMode;
+}
+
+export default function VerticalTagInputItem({item, onClick, onChange, onDelete}) {
+    const displayIndex = getDisplayIndex(item);
+
+    return (
+        <div className={"tag-item-block"}>
+            <li className={getItemClassName(item)} style={{
+                borderColor: item.tag.color,
+                background: item.tag.color,
+            }}>
+                <div
+                    className={"tag-color"}
+                    onClick={(e) => onClick(e, {clickTarget: TagEditMode.Color})}
+                    style={getColorStyle(item)}
+                ></div>
+                <div
+                    className={"tag-content"}
+                    onClick={(e) => onClick(e, {clickTarget: TagEditMode.Name})}
+                >
+                    {getTagContent(item, onChange, onDelete)}
+                </div>
+                {
+                    (displayIndex !== null) &&
+                    <div className={"tag-index"}>
+                        [{displayIndex}]
+                    </div>
+                }
+
+            </li>
+            {
+                (item.isBeingEdited && item.tagEditMode === TagEditMode.Color) ? getColorPicker(item, onChange) : ""
+            }
+        </div>
+    );
+}
+
+function getItemClassName(item) {
+    let className = "tag-item";
+    if (item.isSelected) {
+        className += " tag-item-selected";
+    }
+    return className;
+}
+
+function getColorPicker(item, onChange) {
+    return (
+        <div className="tag-color-picker">
+            <GithubPicker
+                color={item.tag.color}
+                onChangeComplete={(color) => handleColorEdit(item.tag, color, onChange)}
+                colors={tagColors}
+                width={165}
```

Comment:
should these styles/magic number live in our CSS or as consts somewhere else?

Reply:
We do have a constants file, that would probably be a good place for them